Assets/MRTK/Core/Definitions/Devices/MixedRealityControllerVisualizationProfile.cs

```diff
@@ -154,24 +154,33 @@ public GameObject GlobalRightHandVisualizer
         /// </summary>
         public MixedRealityControllerVisualizationSetting[] ControllerVisualizationSettings => controllerVisualizationSettings;
 
-        /// <summary>
-        /// Gets the override model for a specific controller type and hand
-        /// </summary>
-        /// <param name="controllerType">The type of controller to query for</param>
-        /// <param name="hand">The specific hand assigned to the controller</param>
-        public GameObject GetControllerModelOverride(Type controllerType, Handedness hand)
+        private MixedRealityControllerVisualizationSetting? GetControllerVisualizationDefinition(Type controllerType, Handedness hand)
         {
             for (int i = 0; i < controllerVisualizationSettings.Length; i++)
             {
                 if (SettingContainsParameters(controllerVisualizationSettings[i], controllerType, hand))
                 {
-                    return controllerVisualizationSettings[i].OverrideControllerModel;
+                    return controllerVisualizationSettings[i];
                 }
             }
-
             return null;
         }
 
+        public bool ControllerVisualizationDefinitionExists(Type controllerType, Handedness hand)
+        {
+            return GetControllerVisualizationDefinition(controllerType, hand).HasValue;
+        }
+
+        /// <summary>
+        /// Gets the override model for a specific controller type and hand
+        /// </summary>
+        /// <param name="controllerType">The type of controller to query for</param>
+        /// <param name="hand">The specific hand assigned to the controller</param>
+        public GameObject GetControllerModelOverride(Type controllerType, Handedness hand)
+        {
+            return GetControllerVisualizationDefinition(controllerType, hand).HasValue ? GetControllerVisualizationDefinition(controllerType, hand).Value.OverrideControllerModel : null;
```

Comment:
Can the methods throughout here cache the result of `GetControllerVisualizationDefinition(controllerType, hand)` first so it isn't called twice (and thus iterating through all vis settings and checking them twice)?

Reply:
I was thinking that, though out number of visualization settings (less than 10) is low enough that I think the costs of spinning up a hashtable might be higher 😅 given that this is only called during TryGetRenderControllerModels

Reply:
> that I think the costs of spinning up a hashtable might be higher

I'm not sure I know what this hashtable is referring to. I was thinking something like 

```suggestion
            MixedRealityControllerVisualizationSetting? setting = GetControllerVisualizationDefinition(controllerType, hand);
            return setting.HasValue ? setting.Value.OverrideControllerModel : null;
```

Reply:
ooohhhh I thought you meant storing the resulting definition from the GetControllerVisualizationDefinition(controllerType, hand) call and saving it ala. dynamic programming.

This simpler fix I can do and makes a lot more sense lol